xclim/indices/_agro.py

```diff
@@ -113,3 +97,65 @@ def corn_heat_units(
 
     chu.attrs["units"] = ""
     return chu
+
+
+@declare_units(
+    tasmin="[temperature]",
+    tasmax="[temperature]",
+    thresh_tasmin="[temperature]",
+    thresh_tasmax="[temperature]",
+)
+def biologically_effective_degree_days(
+    tasmin: xarray.DataArray,
+    tasmax: xarray.DataArray,
+    thresh_tasmin: str = "10 degC",
+    thresh_tasmax: str = "19 degC",
+    lat_dim: str = "lat",
+    start_date: str = "4",
+    end_date: str = "10",
```

Comment:
```suggestion
    lat : xr.DataArray,
    thresh_tasmin: str = "10 degC",
    thresh_tasmax: str = "19 degC",
    start_date: DayOfYearStr = "04-01",
    end_date: DayOfYearStr = "10-01",
    freq: str = "YS"
```
So that `lat` needs not be a dimension. (think unstructured or rotated pole grids).

`DayOfYearStr` is recognized by the `Indicator` constructor.

I would have a `freq` argument still. In part because the Indicator class behaves differently wether it is present or not and in part to leave all these decisions to the user.

There is the `allowed_periods` attribute that can be given to the Indicator constructor if you want to restrict to yearly. (`allowed_periods=['A']`).

Reply:
Thanks for the comments! 

For the end date, I believe it goes until "10-31", so this could complicate 360-day calendars. Not sure how best to proceed.

Typically, this type of indice is only valid at the yearly scale. I don't see how this info would be useful at the `'M'` frequency, but I don't see the harm in opening it up.

Reply:
Thing is, the missing value checks need to know `freq` to run, which is one of the reason to leave them in the signature. Also, there still a need for the "YS" <-> "AS-JUL" distinction!

Would "11-01" work? If needed, a "endpoint=True/False" could be added to `aggregate_between_dates`?

Reply:
> Thing is, the missing value checks need to know `freq` to run, which is one of the reason to leave them in the signature. Also, there still a need for the "YS" <-> "AS-JUL" distinction!

Sounds good. If we can handle North and South hemisphere, all the better.

> Would "11-01" work? If needed, a "endpoint=True/False" could be added to `aggregate_between_dates`?

I think that would work, yes. I can think of two other indices (not presently coded, but easily added) that could benefit from this functionality as well.